### PR TITLE
feat: authz'd scoped requests for Katsu public endpoints

### PR DESCRIPTION
**Pull request by davidlougheed**
also updates dependencies

replaces #173

### CODE REVIEW

File: src/js/features/ingestion/lastIngestion.store.ts [hunk 1]
```diff
@@ -2,14 +2,22 @@ import type { PayloadAction } from '@reduxjs/toolkit';
...
+  LastIngestionDataTypeResponse[],
+  void,
+  {
+    rejectValue: string;
+    state: RootState;
+  }
+>('dataTypes/fetchKatsuData', (_, { rejectWithValue, getState }) =>
   axios
-    .get(katsuLastIngestionsUrl)
+    .get(katsuLastIngestionsUrl, authorizedRequestConfig(getState()))
```
Comment (v-rocheleau) [inline src/js/features/ingestion/lastIngestion.store.ts:20]: Not `scopedAuthorizedRequestConfig`?

File: src/js/features/search/makeGetSearchFields.thunk.ts [hunk 2]
```diff
@@ -11,7 +12,10 @@ export const makeGetSearchFields = createAsyncThunk<
   { rejectValue: string; state: RootState }
 >('query/makeGetSearchFields', async (_, { rejectWithValue, getState }) => {
   return await axios
-    .get(searchFieldsUrl, { params: getState().metadata.selectedScope })
+    .get(searchFieldsUrl, {
+      headers: { ...makeAuthorizationHeader(getState().auth.accessToken) },
+      params: getState().metadata.selectedScope,
```
Comment (v-rocheleau) [inline src/js/features/search/makeGetSearchFields.thunk.ts:17]: Isn't this the same as using `scopedAuthorizedRequestConfig`?

File: src/js/features/search/makeGetKatsuPublic.thunk.ts [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
 export const makeGetKatsuPublic = createAsyncThunk<
   KatsuSearchResponse,
   void,
   { state: RootState; rejectValue: string }
->('query/makeGetKatsuPublic', (_ignore, thunkAPI) => {
-  const queryParams = thunkAPI.getState().query.queryParams;
+>('query/makeGetKatsuPublic', (_, { rejectWithValue, getState }) => {
+  const state = getState();
+  const scopeParams = state.metadata.selectedScope;
+  const queryParams = { ...scopeParams, ...state.query.queryParams };
```
Comment (v-rocheleau) [inline src/js/features/search/makeGetKatsuPublic.thunk.ts:16]: Rename to something like `scopedQueryParams`, since it includes both.

Comment (v-rocheleau) [reply src/js/features/search/makeGetKatsuPublic.thunk.ts:16]: We could also modify `scopedAuthorizedRequestConfig` to have a `useQuery` boolean argument that tells the function to include `state.query.queryParams`.

**Review by v-rocheleau — CHANGES_REQUESTED** [review]

File: src/js/features/dataTypes/dataTypes.store.ts [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
-  return data;
+export const makeGetDataTypes = createAsyncThunk<
+  object,
+  void,
+  {
+    rejectValue: string;
+    state: RootState;
+  }
+>('dataTypes/makeGetDataTypes', async (_, { getState }) => {
+  const res = await axios.get('/api/service-registry/data-types', authorizedRequestConfig(getState()));
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/js/features/dataTypes/dataTypes.store.ts:17]: this is just a collection of all possible data types in the instance, so yeah it doesn't need scoping

[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/js/features/dataTypes/dataTypes.store.ts:17]: although maybe with counts it makes sense to scope it... TBD I guess? I'll add a note

File: src/js/features/ingestion/lastIngestion.store.ts (hunk 1 above)
Comment (davidlougheed) [reply src/js/features/ingestion/lastIngestion.store.ts:20]: good point.

File: src/js/features/dataTypes/dataTypes.store.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (v-rocheleau) [reply src/js/features/dataTypes/dataTypes.store.ts:17]: Yeah I was thinking about the counts here

File: src/js/features/search/makeGetSearchFields.thunk.ts (hunk 2 above)
Comment (davidlougheed) [reply src/js/features/search/makeGetSearchFields.thunk.ts:17]: yes, forgot during a refactor.

File: src/js/features/search/makeGetKatsuPublic.thunk.ts (hunk 3 above)
Comment (davidlougheed) [reply src/js/features/search/makeGetKatsuPublic.thunk.ts]: added an extra argument to `scopedAuthorizedRequestConfig` for arbitrary extra query params

File: src/js/features/search/makeGetSearchFields.thunk.ts (hunk 2 above)
Comment (davidlougheed) [reply src/js/features/search/makeGetSearchFields.thunk.ts]: done

File: src/js/features/dataTypes/dataTypes.store.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/js/features/dataTypes/dataTypes.store.ts:20]: added a note for now, will open a redmine ticket

[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/js/features/dataTypes/dataTypes.store.ts:20]: https://redmine.c3g-app.sd4h.ca/issues/2297

**Pull request merged**